### ISSUE TITLE
Core crashing retroarch on content close.

### ISSUE THREAD

**Issue by stuken** [libretro/melonDS#124]
As per title, after the 0.9.3 rebase the core is now crashing out on content close.

I cant get any further debugging as debug builds dont seem to work, printf and fflush(stdout) causes compile errors and there is a nice exception storm on startup. But based on the last stdout line thats printed it at least gets to https://github.com/libretro/melonDS/blob/cba7ecce4aadcf804645b646f0522926d85144ec/src/ARMJIT.cpp#L1145 during the shutdown process.

```
#0  0x00007ffd488af123 in ntdll!RtlIsZeroMemory ()
   from C:\Windows\SYSTEM32\ntdll.dll
No symbol table info available.
#1  0x00007ffd488b7ee2 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SYSTEM32\ntdll.dll
No symbol table info available.
#2  0x00007ffd488b81ca in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SYSTEM32\ntdll.dll
No symbol table info available.
#3  0x00007ffd488bde51 in ntdll!RtlpNtSetValueKey ()
   from C:\Windows\SYSTEM32\ntdll.dll
No symbol table info available.
#4  0x00007ffd48857142 in ntdll!memset () from C:\Windows\SYSTEM32\ntdll.dll
No symbol table info available.
#5  0x00007ffd487d47b1 in ntdll!RtlFreeHeap ()
   from C:\Windows\SYSTEM32\ntdll.dll
No symbol table info available.
#6  0x00007ffd46e39c9c in msvcrt!free () from C:\Windows\System32\msvcrt.dll
No symbol table info available.
#7  0x00007ffcabbd847d in ?? ()
   from C:\Users\Stuart\Documents\GitHub\libretro\RetroArch\cores\melonds_libretro.dll
No symbol table info available.
#8  0x00007ffcabc3fd30 in ?? ()
   from C:\Users\Stuart\Documents\GitHub\libretro\RetroArch\cores\melonds_libretro.dll -> must be NDS::DeInit()
No symbol table info available.
#9  0x00007ffcabbd9476 in ?? ()
   from C:\Users\Stuart\Documents\GitHub\libretro\RetroArch\cores\melonds_libretro.dll -> must be retro_unload_game()
No symbol table info available.
#10 0x00007ff6f3c41e34 in core_unload_game (p_rarch=0x7ff6f4b11ce0 <rarch_st>)
    at retroarch.c:30100
No locals.
```

**Comment by m4tsa** [comment]
Should be fixed by 5885a9be722329e10016ab6f73587fc6e9905621.

**Comment by stuken** [comment]
Confirmed that change has fixed the issue. Thanks.
